include/soci/exchange-traits.h

```diff
@@ -148,6 +152,15 @@ struct exchange_traits<long_string>
     enum { x_type = x_longstring };
 };
 
+#ifdef SOCI_HAVE_BOOST
+template <typename T>
+struct exchange_traits<boost::optional<T> >
+{
+    typedef typename exchange_traits<T>::type_family type_family;
+    enum { x_type = exchange_traits<T>::x_type };
+};
+#endif
+
```

Comment:
Could this be done in `include/soci/boost-optional.h`? It seems like a more appropriate place for it.

Reply:
I try do this